Day-5-Exercises/main.js

```diff
@@ -0,0 +1,217 @@
+?itCompanies.splice(center, 1)
+:itCompanies.splice(center - 1, 2)
+console.log(itCompanies)
+
+// Remove the last IT company from the array
+itCompanies.pop()
+console.log(itCompanies)
+
+// Remove all IT companies
+itCompanies.splice(0,itCompanies.length)
+console.log(itCompanies,'The End')
+
+
+//Exercise: Level 2
+
+//Create a separate countries.js file and store the countries array in to this file, create a separate file web_techs.js and store the webTechs array in to this file. Access both file in main.js file
+
+//First remove all the punctuations and change the string to array and count the number of words in the array
+let text = 'I love teaching and empowering people. I teach HTML, CSS, JS, React, Python.'
+const words = text.replace(/[.,]/g,'').split(' ')
+console.log(words)
+console.log(words.length)
+
+//In the following shopping cart add, remove, edit items
+/*
+add 'Meat' in the beginning of your shopping cart if it has not been already added
+add Sugar at the end of you shopping cart if it has not been already added
+remove 'Honey' if you are allergic to honey
+modify Tea to 'Green Tea'
+*/
+const shoppingCart = ['Milk', 'Coffee', 'Tea', 'Honey']
+if (!shoppingCart.includes('Meat')) {shoppingCart.unshift('Meat')}
+if (!shoppingCart.includes('Sugar')) {shoppingCart.push('Sugar')}
+shoppingCart.splice(shoppingCart.indexOf('Honey'),1)
+shoppingCart[shoppingCart.indexOf('Tea')] = 'Green Tea'
+console.log(shoppingCart)
+
+//In countries array check if 'Ethiopia' exists in the array if it exists print 'ETHIOPIA'. If it does not exist add to the countries list.
+let newCountry = 'Ethiopia'
+countries.includes(newCountry)?console.log(newCountry.toUpperCase()):countries.push(newCountry)
+console.log(countries)
+
+//In the webTechs array check if Sass exists in the array and if it exists print 'Sass is a CSS preprocess'. If it does not exist add Sass to the array and print the array.
+let newTech = 'Sass'
+webTechs.includes(newTech)?console.log(`${newTech} is a CSS preprocess`):webTechs.push(newTech)
+console.log(webTechs)
+
+//Concatenate the following two variables and store it in a fullStack variable.
+const frontEnd = ['HTML', 'CSS', 'JS', 'React', 'Redux']
+const backEnd = ['Node','Express', 'MongoDB']
+
+const fullStack = frontEnd.concat(backEnd)
+
+console.log(fullStack)
+
+
+//Exercise: Level 3
+
+//The following is an array of 10 students ages:
+const ages = [19, 22, 19, 24, 20, 25, 26, 24, 25, 24]
+
+/*
+- Sort the array and find the min and max age
+- Find the median age(one middle item or two middle items divided by two)
+- Find the average age(all items divided by number of items)
+- Find the range of the ages(max minus min)
+- Compare the value of (min - average) and (max - average), use abs() method 
+*/
+ages.sort()
+let minAge = ages[0]
+let maxAge = ages[ages.length - 1]
+console.log(ages)
+console.log( `Min age: ${minAge}, Max age: ${maxAge}`)
+
+let sum = 0
+ages.forEach(element => {sum = sum + element})
```

Comment:
can you try using `.reduce()` method which does the same job?

Reply:
added alternative option via .reduce()